sql/driver/conn.go

```diff
@@ -29,6 +29,7 @@ import (
 type conn struct {
 	sender  Sender
 	session []byte
+	txn     []byte
```

Comment:
Rather than keeping separate `conn.txn` field, can't the transaction info be stored inside the session state?